src/CampaignKit.WorldMap/Controllers/MapController.cs

```diff
@@ -24,289 +24,278 @@
+				return View();
+
+			var map = new Map
+			{
+				Name = model.Name,
+				Secret = model.Secret,
+				Copyright = model.Copyright,
+				ContentType = model.MapImage.ContentType,
+				FileExtension = Path.GetExtension(model.MapImage.FileName ?? string.Empty).ToLower(),
+				CreationTimestamp = DateTime.UtcNow,
+				RepeatMapInX = model.RepeatMapInX
+			};
+
+			var id = await _mapDataService.Create(map, model.MapImage.OpenReadStream());
+			if (id == 0)
+			{
+				ModelState.AddModelError(string.Empty,
+					"Your map could not be saved. Please try again.");
+			}
+			else
+			{
+				return RedirectToAction(nameof(Show), new { id, map.Secret, ShowProgress = true });
+			}
+
+			return View();
+		}
+
+		[HttpGet]
+		public async Task<IActionResult> Delete(int id, string secret)
+		{
+			var model = await _mapDataService.Find(id);
+
+			if (model == null || model.Secret != secret)
+			{
+				return DeleteErrorView();
+			}
+				
+			return View(new MapDeleteViewModel { Name = model.Name, HiddenId = model.MapId, HiddenSecret = model.Secret });
+		}
+
+		[HttpPost]
+		[ValidateAntiForgeryToken]
+		public async Task<IActionResult> Delete(int id, string secret, MapDeleteViewModel model)
+		{
+			if (!ModelState.IsValid)
+				return View();
+
+			var map = await _mapDataService.Find(id);
+
+			if (map == null || map.Secret != secret)
+				return DeleteErrorView();
+
+			if (map.MapId != model.HiddenId || map.Secret != model.HiddenSecret)
+			{
+				ModelState.AddModelError(string.Empty,
+					"Your map could not be deleted. Please try again.");
+			}
+			else
+			{
+				await _mapDataService.Delete(id);
+				return RedirectToAction(nameof(Index));
+			}
+
+			return View();
+		}
+
+		[HttpGet]
+		public async Task<IActionResult> Edit(int id, string secret)
+		{
+			var model = await _mapDataService.Find(id);
+
+			if (model == null || model.Secret != secret) return EditErrorView();
+
+			return View(new MapEditViewModel
+			{
+				Name = model.Name,
+				Copyright = model.Copyright,
+				RepeatMapInX = model.RepeatMapInX
+			});
+
+		}
+
+		[HttpPost]
+		[ValidateAntiForgeryToken]
+		public async Task<IActionResult> Edit(int id, string secret, MapEditViewModel model)
+		{
+			if (!ModelState.IsValid)
+			{
+				return View();
+			}
+
+			var map = await _mapDataService.Find(id);
+
+			if (map == null || map.Secret != secret) return EditErrorView();
+
+			map.Name = model.Name;
+			map.Copyright = model.Copyright;
+			map.RepeatMapInX = model.RepeatMapInX;
+
+			var result = await _mapDataService.Save(map);
+			if (!result)
+				ModelState.AddModelError(string.Empty,
+					"Your map could not be saved. Please try again.");
+			else
+				return RedirectToAction(nameof(Show), new { map.MapId, map.Secret, ShowProgress = true });
+
+			return View();
+		}
+
+		public async Task<IActionResult> Index()
+		{
+			var model = await _mapDataService.FindAll();
+			model.OrderByDescending(m => m.CreationTimestamp);
```

Comment:
Is OrderByDescending() working in place or does it return the sorted model?
```suggestion
 			model = model.OrderByDescending(m => m.CreationTimestamp);
```